module/LearnZF2AjaxImageGallery/src/LearnZF2AjaxImageGallery/Form/AjaxImageUploadForm.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+namespace LearnZF2AjaxImageGallery\Form;
+
+use Zend\Form\Form;
+use Zend\Form\Element;
+
+class AjaxImageUploadForm extends Form
+{
+    public function __construct()
+    {
+        parent::__construct("content");
+        $this->setAttribute('method', 'post');
+
+        $this->add([
+            'name' => 'imageUpload',
+            'type'  => 'File',
+            'attributes' => [
+                'class' => 'imgupload',
+                'id' => 'imgajax',
+                'multiple' => true,
+            ],
+        ]);
+    }

```

Comment:
I have a xModel which implements InputFilterAwareInterface with get|setInputFIlter() and I have a form which implements InputFilterProviderInterface with init() and getInputFilterSpecification() functions. Both input filters validates the form according to their interface functions. Which one should I use? Should I leave them like that, or I can move the getInputFilterSpecification into the xModel class.